src/domain/prices/prices.repository.ts

```diff
@@ -0,0 +1,63 @@
+import { Inject, Injectable } from '@nestjs/common';
+import {
+  ILoggingService,
+  LoggingService,
+} from '../../logging/logging.interface';
+import { IPricesApi } from '../interfaces/prices-api.interface';
+import { IPricesRepository } from './prices.repository.interface';
+import { isArray } from 'lodash';
+
+@Injectable()
+export class PricesRepository implements IPricesRepository {
+  constructor(
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+    @Inject(IPricesApi) private readonly pricesApi: IPricesApi,
+  ) {}
+
+  async getNativeCoinPrice(args: {
+    nativeCoinId: string;
+    fiatCode: string;
+  }): Promise<number> {
+    try {
+      const result = await this.pricesApi.getNativeCoinPrice(args);
+      const { nativeCoinId, fiatCode } = args;
+      const nativeCoinPrice = result?.[nativeCoinId]?.[fiatCode];
+      if (!nativeCoinPrice) {
+        this.loggingService.error(`Got an invalid ${nativeCoinId} price`);
+        return 0;
```

Comment:
I would throw an error in these cases. `0` should be seen as a valid value from the domain's perspective.

Mapping the error to `0` can be done on the route level because there we have the contract with the clients (API). But this is not the case on the domain layer and it's better to build a more meaningful result at this level.

Reply:
It could be both cases. The provider is not helping here because the status codes and the payloads are not very helpful. The status code is always 200, no matter the kind of absence (token doesn't exist, or fiat code is not supported). Examples:
- Token doesn't exist (made up address): https://api.coingecko.com/api/v3/simple/token_price/ethereum?contract_addresses=0xa0b86991c6218b36c1d19d4b2e9eb0ce3606eb48&vs_currencies=eur
- Native coin doesn't exist: https://api.coingecko.com/api/v3/simple/price?ids=foo&vs_currencies=eur
- Token exists but fiat code is not supported: https://api.coingecko.com/api/v3/simple/token_price/ethereum?contract_addresses=0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48&vs_currencies=foo

So for the sake of simplicity, I'm just inspecting the response and throwing an error if it doesn't contain the price. Of course, we could implement some logic to distinguish between those scenarios and throw ad-hoc errors. That could make the code a bit more complex but I'd be happy to add this if you think it's valuable.

Reply:
I've changed this to return `null` from the domain layer when an error happens in [17040f3](https://github.com/safe-global/safe-client-gateway/pull/732/commits/17040f3f07ad63a3682ce6b665dc451dfae545a2)